include/picongpu/particles/debyeLength/Check.hpp

```diff
@@ -0,0 +1,180 @@
+                        mapper,
+                        hostDeviceBuffer.getDeviceBuffer().getDataBox());
+                    hostDeviceBuffer.deviceToHost();
+
+                    // Copy is asynchronous, need to wait for it to finish
+                    __getTransactionEvent().waitForFinished();
+                    dc.releaseData(Frame::getName());
+                    return hostBox(0);
+                }
+
+                //! Result of the Debye length check
+                enum class Status
+                {
+                    Passed,
+                    Failed,
+                    Skipped
+                };
+
+                /** Check Debye length resolution for the given electron species
+                 *  in the global simulation volume
+                 *
+                 * This function must be called from all MPI ranks.
+                 *
+                 * @tparam T_ElectronSpecies electron species type
+                 *
+                 * @param cellDescription mapping for kernels
+                 */
+                template<typename T_ElectronSpecies>
+                HINLINE Status checkDebyeLength(MappingDesc const cellDescription)
+                {
+                    auto localResult = checkLocalDebyeLength<T_ElectronSpecies>(cellDescription);
+                    auto globalResult = Result{};
+                    pmacc::mpi::MPIReduce reduce;
+                    reduce(
+                        pmacc::nvidia::functors::Add(),
+                        &globalResult.numViolatingSupercells,
+                        &localResult.numViolatingSupercells,
+                        1,
+                        pmacc::mpi::reduceMethods::AllReduce());
+                    reduce(
+                        pmacc::nvidia::functors::Add(),
+                        &globalResult.numActiveSupercells,
+                        &localResult.numActiveSupercells,
+                        1,
+                        pmacc::mpi::reduceMethods::AllReduce());
+                    if(globalResult.numActiveSupercells)
+                    {
+                        auto const ratioFailing = static_cast<float_64>(globalResult.numViolatingSupercells)
+                            / static_cast<float_64>(globalResult.numActiveSupercells);
+                        /* Due to random nature of temperature and estimates used internally,
+                         * we allow a relatively small number of supercells to fail the check
+                         */
+                        auto const thresholdFailing = 0.1;
+                        if(ratioFailing <= thresholdFailing)
+                            return Status::Passed;
+                        else
+                            return Status::Failed;
+                    }
+                    else
+                        return Status::Skipped;
+                }
+
+            } // namespace detail
+
+            /** Check Debye length resolution
+             *
```

Comment:
Please add a line or two describing what the actual check/limit is. With a reference, please (e.g. Hockney, p. 32; Birdsall p. 389).

Reply:
@sbastrakov I think you missed this comment in the past.

Reply:
please check this comment: https://github.com/ComputationalRadiationPhysics/picongpu/pull/3446/files#r537676156

Reply:
done